### DISCUSSION TITLE
How to wait for a delete?

### DISCUSSION THREAD

**Discussion by degeens** [kubernetes-client/csharp#870] [Q&A]
Hi all 👋

I'm looking for a nice way to delete an existing Kubernetes job and then create a new job with the same name. This is a (simplified) code snippet that I came up with:

```c#
var job = new V1Job { ... };
var @namespace = "default";

await Client.DeleteNamespacedJobAsync(job.Metadata.Name, @namespace)
await Client.CreateNamespacedJobAsync(job, @namespace);
```

However, when I execute this code I will always end up with the following exception: "object is being deleted: jobs.batch "..." already exists". It looks like the task that is returned by DeleteNamespacedJobAsync completes early before the delete is done.

What is the cleanest possible way to solve this using the Kubernetes Client library?

Thanks for the help!

**Comment by tg123** [comment]
see #680 

you can also take a look at the `cleanup` in e2e test
https://github.com/kubernetes-client/csharp/blob/e45fa773bade29c37f00198c701543b6c2c122c7/tests/E2E.Tests/MnikubeTests.cs#L29

**Comment by degeens** [comment]
Thank you for your reply.

Could you provide some more information about the `watch` callback you are talking about in #680? Where can I find it?

**Comment by brendandburns** [comment]
There is a watch callback example here:
https://github.com/kubernetes-client/csharp/blob/master/examples/watch/Program.cs#L32

You will get a delete event when the object is deleted.

**Comment by degeens** [comment]
Thank you for your help! I'm closing the issue.
